src/engine/qcommon/qcommon.h

```diff
@@ -285,7 +285,7 @@ You or the server may be running older versions of the game."
 
 #define PROTOCOL_VERSION       86
 
-#define URI_SCHEME             GAMENAME_STRING "://"
+#define URI_SCHEME             va("%s://", current_game->gamename_string)
```

Comment:
Does the protocol have to optionally announce the game?
like¹ `dgp://gg.illwieckz.net:27960/unvanquished`

That `/unvanquished` would be optional, but if you have multiple games running dæmon engine, your game manager can use the binary (xonotic for `/xonotic`, unvanquished for `/unvanquished`, would fill the `-game` switch), if game not installed but daemon from another game installed, run it with that, if that optional `/game` is not there, same. That would help Dæmon based games with some outdated engine to not break. That would be very useful for games that enables `fs_legacypaks` by default (like xonotic), ensuring a game client enabling that legacy stuff is started before joining server.

¹ `dgp://` for _Daemon Game Protocol_ for example or another variation with _Connect Protocol_ or words like that, meaning it's a protocol to make game clients connecting a server, following _Dæmon guidelines_, the same way other engine can reuse the _DPK_ guidelines because they are tired of PK3 design flaws if they want. If defined well, other games would be able to reuse the url format even if they do not rely on Dæmon engine. Well, that's a new topic we can have in forum so I will probably not elaborate more about that topic on that thread. :grin: